src/main/java/lotto/service/lotto/DefaultLottoMachine.java

```diff
@@ -0,0 +1,57 @@
+package lotto.service.lotto;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import lotto.dto.MatchingResultDto;
+import lotto.model.lotto.lottoNumber.NumberPicker;
+import lotto.model.lotto.winningResult.rank.RankDeterminer;
+import lotto.model.winningNumber.BonusNumber;
+import lotto.model.lotto.lottoNumber.Lotto;
+import lotto.model.lotto.purchaseAmount.PurchaseAmount;
+import lotto.model.lotto.lottoNumber.Lottos;
+import lotto.model.winningNumber.MainNumber;
+import lotto.model.lotto.winningResult.rank.Rank;
+import lotto.model.lotto.winningResult.WinningResults;
+
+public class DefaultLottoMachine implements LottoMachine {
+    private static final int PERCENTAGE_MULIPLIER = 100;
+
+    private final RankDeterminer rankDeterminer;
+    private final NumberPicker numberPicker;
+
+    public DefaultLottoMachine(RankDeterminer rankDeterminer, NumberPicker numberPicker) {
+        this.rankDeterminer = rankDeterminer;
+        this.numberPicker = numberPicker;
+    }
+
+    @Override
+    public Lottos issueLottos(PurchaseAmount purchaseAmount) {
+        List<Lotto> lottos = IntStream.range(0, purchaseAmount.calculateLottoAmount())
+                .mapToObj(i -> new Lotto(numberPicker.pick()))
+                .collect(Collectors.toList());
```

Comment:
p3

```suggestion
        List<Lotto> lottos = IntStream.range(0, purchaseAmount.calculateLottoAmount())
                .mapToObj(i -> new Lotto(numberPicker.pick()))
                .toList();
```

toList()로 반환해보시는 것도 한 번 생각해주시면 감사합니다. 

그 이유는 아래와 같습니다.  

`issueLottos` 메서드는 lottos를 최종적으로 `new Lottos()`에 넣고 반환합니다.  
Lottos 클래스의 내부 구현을 보면 lottos를 불변 list로 만들고자 하는 의도가 보입니다.

```java
@Override
    public List<Lotto> lottos() {
        return Collections.unmodifiableList(lottos);
    }
```

따라서 처음부터 Lottos 클래스에 전달하기 전에 `.toList()`를 사용해서 불변 List를 전달해주는 것도 하나의 방법이 될 것 같습니다.  

이렇게 변경을 한다면 코드가 조금 더 간결해지고 가독성도 조금 상승이되면서 우리가 자바 21을 사용하는 이유도 부각시킬 수 있을 것 같습니다. `.collect(Collectors.toList())`는 자바 8 문법이니까용

Reply:
엇 이부분은 처음 알았네요..! 
아직 스트림을 잘 몰라서 알음알음 써보는데 이건 처음 알았습니다 ㅎㅎ
감사해요! 바로 적용해볼게요!!